src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -1188,5 +1184,86 @@ private void SearchInCurrentPage(in EncodedKey key, ref CursorState state)
                 mid++;
             state.LastSearchPosition = ~mid;
         }
+
+        private static int DictionaryOrder(ReadOnlySpan<byte> s1, ReadOnlySpan<byte> s2)
+        {
+            // Bed - Tree: An All-Purpose Index Structure for String Similarity Search Based on Edit Distance
+            // https://event.cwi.nl/SIGMOD-RWE/2010/12-16bf4c/paper.pdf
+
+            //  Intuitively, such a sorting counts the total number of strings with length smaller than |s|
+            //  plus the number of string with length equal to |s| preceding s in dictionary order.
+
+            int len1 = s1.Length;
+            int len2 = s2.Length;
+
+            if (len1 == 0 && len2 == 0)
+                return 0;
+
+            if (len1 == 0)
+                return -1;
+
+            if (len2 == 0)
+                return 1;
+
+            //  Given two strings si and sj, it is sufficient to find the most significant position p where the two string differ
+            int minLength = len1 < len2 ? len1 : len2;
+
+            // If π(si[p]) < π(sj[p]), we can assert that si precedes sj in dictionary order φd, and viceversa.
+            int i;
+            for (i = 0; i < minLength; i++)
+            {
+                if (s1[i] < s2[i])
+                    return -1;
+                else if (s2[i] < s1[i])
+                    return 1;
+            }
+
+            if (len1 < len2)
+                return -1;
+
+            return 1;
+        }
+
+        private EncodedKey FuzzySearchPageAndPushNext(in EncodedKey key, ref IteratorCursorState cstate)
```

Comment:
I don't understand what you are doing here.

Reply:
Good catch, in the end it is exactly that the original. Before I fixed a few errors on the algorithm it was actually different.